File: scripts/migrate_types.py
#!/usr/bin/env python3
import os
import re
from pathlib import Path

PROJECT_ROOT = Path(__file__).parent.parent
CLIENT_SRC = PROJECT_ROOT / 'client' / 'src'

# Define import mappings
REPLACEMENTS = {
    r"from\s+['\"]@client/types['\"]": "from '@client/shared/types'",
    r"from\s+['\"]@client/types/dashboard['\"]": "from '@client/shared/types/dashboard'",
    r"from\s+['\"]@client/types/loading['\"]": "from '@client/shared/types/loading'",
    r"from\s+['\"]@client/types/navigation['\"]": "from '@client/shared/types/navigation'",
    r"from\s+['\"]@client/types/mobile['\"]": "from '@client/shared/types/mobile'",
    r"from\s+['\"]@client/types/user-dashboard['\"]": "from '@client/shared/types/user-dashboard'",
    r"from\s+['\"]@client/types/onboarding['\"]": "from '@client/features/users/types'",
    r"from\s+['\"]@client/types/expert['\"]": "from '@client/features/users/types'",
    r"from\s+['\"]@client/types/community['\"]": "from '@client/features/community/types'",
    r"from\s+['\"]@client/types/conflict-of-interest['\"]": "from '@client/features/analysis/types'",
    r"from\s+['\"]@client/types/auth['\"]": "from '@client/core/auth'",
    r"from\s+['\"]@client/types/realtime['\"]": "from '@client/core/realtime/types'",
}

def process_file(filepath):
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()

        original = content
        for pattern, replacement in REPLACEMENTS.items():
            content = re.sub(pattern, replacement, content)

        if content != original:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(content)
            return True, 1
        return False, 0
    except Exception as e:
        print(f"Error processing {filepath}: {e}")
        return False, 0

def main():
    print("🔄 Starting type import migration...\n")

    ts_files = list(CLIENT_SRC.rglob('*.ts')) + list(CLIENT_SRC.rglob('*.tsx'))
    print(f"📁 Found {len(ts_files)} TypeScript files\n")

    updated = 0
    total_changes = 0

    for filepath in ts_files:
        if '@client/types' in open(filepath, 'r', encoding='utf-8', errors='ignore').read():
            changed, count = process_file(filepath)
            if changed:
                updated += 1
                total_changes += count
                relpath = filepath.relative_to(CLIENT_SRC)
                print(f"✓ {relpath}")

    print(f"\n✅ Complete!\n")
    print(f"📊 Files updated: {updated}")
    print(f"📊 Total changes: {total_changes}")

if __name__ == '__main__':
    main()

File: scripts/migrate-types.js
#!/usr/bin/env node

/**
 * Comprehensive Type Import Migration Script
 * Migrates all imports from client/src/types/ to appropriate FSD locations
 */

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const PROJECT_ROOT = path.join(__dirname, '..');
const CLIENT_SRC = path.join(PROJECT_ROOT, 'client', 'src');

// Define comprehensive import mappings
const IMPORT_MAPPINGS = [
  // Feature-specific types
  {
    pattern: /from\s+['"](\.\.?\/)*types\/community['"];/g,
    replacement: "from '@client/features/community/types';",
    description: 'Community types'
  },
  {
    pattern: /from\s+['"](\.\.?\/)*types\/expert['"];/g,
    replacement: "from '@client/features/users/types';",
    description: 'Expert types'
  },
  {
    pattern: /from\s+['"](\.\.?\/)*types\/onboarding['"];/g,
    replacement: "from '@client/features/users/types';",
    description: 'Onboarding types'
  },
  {
    pattern: /from\s+['"](\.\.?\/)*types\/conflict-of-interest['"];/g,
    replacement: "from '@client/features/analysis/types';",
    description: 'Conflict of interest types'
  },

  // Shared types
  {
    pattern: /from\s+['"](\.\.?\/)*types\/navigation['"];/g,
    replacement: "from '@client/lib/types/navigation';",
    description: 'Navigation types'
  },
  {
    pattern: /from\s+['"](\.\.?\/)*types\/mobile['"];/g,
    replacement: "from '@client/lib/types/mobile';",
    description: 'Mobile types'
  },
  {
    pattern: /from\s+['"](\.\.?\/)*types\/user-dashboard['"];/g,
    replacement: "from '@client/lib/types/user-dashboard';",
    description: 'User dashboard types'
  },

  // Core auth types
  {
    pattern: /from\s+['"](\.\.?\/)*types\/auth['"];/g,
    replacement: "from '@client/core/auth';",
    description: 'Auth types'
  },

  // Core realtime types
  {
    pattern: /from\s+['"](\.\.?\/)*types\/realtime['"];/g,
    replacement: "from '@client/core/realtime/types';",
    description: 'Realtime types'
  },

  // Generic/core types - keep in shared
  {
    pattern: /from\s+['"](\.\.?\/)*types\/core['"];/g,
    replacement: "from '@client/lib/types';",
    description: 'Core types'
  },

  // Fallback for @client/types path alias imports
  {
    pattern: /from\s+['"]@client\/types\/community['"];/g,
    replacement: "from '@client/features/community/types';",
    description: '@client/types/community alias'
  },
  {
    pattern: /from\s+['"]@client\/types\/expert['"];/g,
    replacement: "from '@client/features/users/types';",
    description: '@client/types/expert alias'
  },
  {
    pattern: /from\s+['"]@client\/types\/conflict-of-interest['"];/g,
    replacement: "from '@client/features/analysis/types';",
    description: '@client/types/conflict-of-interest alias'
  },
  {
    pattern: /from\s+['"]@client\/types\/navigation['"];/g,
    replacement: "from '@client/lib/types/navigation';",
    description: '@client/types/navigation alias'
  },
  {
    pattern: /from\s+['"]@client\/types\/mobile['"];/g,
    replacement: "from '@client/lib/types/mobile';",
    description: '@client/types/mobile alias'
  },
  {
    pattern: /from\s+['"]@client\/types\/user-dashboard['"];/g,
    replacement: "from '@client/lib/types/user-dashboard';",
    description: '@client/types/user-dashboard alias'
  },
  {
    pattern: /from\s+['"]@client\/types\/auth['"];/g,
    replacement: "from '@client/core/auth';",
    description: '@client/types/auth alias'
  },
  {
    pattern: /from\s+['"]@client\/types\/realtime['"];/g,
    replacement: "from '@client/core/realtime/types';",
    description: '@client/types/realtime alias'
  },
  {
    pattern: /from\s+['"]@client\/types\/core['"];/g,
    replacement: "from '@client/lib/types';",
    description: '@client/types/core alias'
  },
];

function getAllTypeFiles() {
  const files = [];
  function walk(dir) {
    try {
      const entries = fs.readdirSync(dir, { withFileTypes: true });
      for (const entry of entries) {
        const fullPath = path.join(dir, entry.name);
        if (entry.isDirectory()) {
          if (!entry.name.startsWith('.') && entry.name !== 'node_modules') {
            walk(fullPath);
          }
        } else if ((entry.name.endsWith('.ts') || entry.name.endsWith('.tsx')) && !entry.name.endsWith('.d.ts')) {
          files.push(fullPath);
        }
      }
    } catch (err) {
      console.error(`Error walking ${dir}:`, err.message);
    }
  }
  walk(CLIENT_SRC);
  return files;
}

function updateFile(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf-8');
    let originalContent = content;
    let changeCount = 0;

    for (const mapping of IMPORT_MAPPINGS) {
      const matches = content.match(mapping.pattern);
      if (matches) {
        content = content.replace(mapping.pattern, mapping.replacement);
        changeCount += matches.length;
      }
    }

    if (content !== originalContent) {
      fs.writeFileSync(filePath, content, 'utf-8');
      return { updated: true, changes: changeCount };
    }
    return { updated: false, changes: 0 };
  } catch (err) {
    console.error(`Error processing ${filePath}:`, err.message);
    return { updated: false, changes: 0, error: err.message };
  }
}

function main() {
  console.log('🚀 Starting type import migration...\n');

  const files = getAllTypeFiles();
  console.log(`Found ${files.length} TypeScript files to process\n`);

  let totalFiles = 0;
  let totalChanges = 0;
  const updatedFiles = [];

  for (const filePath of files) {
    const result = updateFile(filePath);
    if (result.updated) {
      totalFiles++;
      totalChanges += result.changes;
      const relPath = path.relative(CLIENT_SRC, filePath);
      updatedFiles.push({ file: relPath, changes: result.changes });
    }
  }

  console.log('\n✅ Migration Complete!\n');
  console.log(`📊 Summary:`);
  console.log(`   Files updated: ${totalFiles}`);
  console.log(`   Total replacements: ${totalChanges}\n`);

  if (updatedFiles.length > 0) {
    console.log('📝 Updated files:');
    updatedFiles.forEach(({ file, changes }) => {
      console.log(`   • ${file} (${changes} changes)`);
    });
  }

  return totalFiles > 0 ? 0 : 1;
}

main();
